src/modules/analytics/utils.ts

```diff
@@ -107,3 +111,136 @@ export function getAnonymousId() {
     return undefined
   }
 }
+
+export function hasEvmWallet() {
+  const userAgent = window.navigator.userAgent
+
+  const isBot = isbot(userAgent)
+  if (isBot) {
+    return undefined
+  }
+
+  return window.ethereum !== undefined
+}
+
+export function getEvmWallets() {
+  const userAgent = window.navigator.userAgent
+
+  const isBot = isbot(userAgent)
+  if (isBot) {
+    return undefined
+  }
+  const windowWallets = window as WindowWithWallets
+  const ethereum = windowWallets.ethereum
+  if (!ethereum) return []
+
+  const names = new Set<string>()
+
+  if (ethereum.isRabby) names.add('rabby')
+  else if (ethereum.isMetaMask) names.add('metamask')
+  if (ethereum.isCoinbaseWallet) names.add('coinbase')
+
+  return Array.from(names)
+}
+
+export function hasSolanaWallet() {
+  const userAgent = window.navigator.userAgent
+
+  const isBot = isbot(userAgent)
+  if (isBot) {
+    return undefined
+  }
+
+  const windowWallets = window as WindowWithWallets
+
+  return (
+    windowWallets.solana !== undefined ||
+    windowWallets.sollet !== undefined ||
+    windowWallets.braveSolana?.isBraveWallet === true ||
+    windowWallets.solong !== undefined ||
+    windowWallets.exodus?.solana !== undefined ||
+    Boolean(windowWallets.glowSolana) ||
+    windowWallets.coin98 !== undefined ||
+    windowWallets.coinbaseSolana !== undefined ||
+    windowWallets.clover_solana !== undefined ||
+    windowWallets.Slope !== undefined ||
+    windowWallets.huobiWallet?.isHuobiWallet === true ||
+    typeof windowWallets?.nightly?.solana !== 'undefined'
```

Comment:
What do you think about doing:
```suggestion
    windowWallets.solana !== undefined || getSolanaWallets().length > 0
```